test/user-test.js

```diff
@@ -0,0 +1,183 @@
+        let user = app.models.User;
+        user.destroyAll({ 'username': 'user1' },
+            (err, obj, count) => {
+                if (err) throw err;
+            });
+
+        user.destroyAll({ 'username': 'userTest' },
+            (err, obj, count) => {
+                if (err) throw err;
+            });
+    });
+
+    it('Empty Username Return Error', (done) => {
+        request(apiAddress)
+            .post('/users')
+            .set('Content-Type', 'application/json')
+            .set('Accept', 'application/json')
+            .send({
+                email: 'user1@gmail.com',
+                password: 'user1',
+                username: ''
+            })
+            .then(res => {
+                expect(res).to.have.status(422);
+                done();
+            })
+            .catch(err => {
+                expect(err).to.not.be.null;
+                expect(err).to.have.status(422);
+                done();
+            });
+    });
+
+    it('Empty Email Return Error', (done) => {
+        request(apiAddress)
+            .post('/users')
+            .set('Content-Type', 'application/json')
+            .set('Accept', 'application/json')
+            .send({
+                email: '',
+                password: 'user1',
+                username: 'user1'
+            })
+            .then(res => {
+                expect(res).to.have.status(422);
+                done();
+            })
+            .catch(err => {
+                expect(err).to.not.be.null;
+                expect(err).to.have.status(422);
+                done();
+            });
+    });
+
+    it('Empty Password Return Error', (done) => {
+        request(apiAddress)
+            .post('/users')
+            .set('Content-Type', 'application/json')
+            .set('Accept', 'application/json')
+            .send({
+                email: 'user1@gmail.com',
+                password: '',
+                username: 'user1'
+            })
+            .then(res => {
+                expect(res).to.have.status(422);
+                done();
+            })
+            .catch(err => {
+                expect(err).to.not.be.null;
+                expect(err).to.have.status(422);
+                done();
+            });
+    });
+
+    it('Invalid Email Format Return Error', (done) => {
+        request(apiAddress)
+            .post('/users')
+            .set('Content-Type', 'application/json')
+            .set('Accept', 'application/json')
+            .send({
+                email: 'user1',
+                password: 'user1',
+                username: 'user1'
+            })
+            .then(res => {
+                expect(res).to.have.status(422);
+                done();
+            })
+            .catch(err => {
+                expect(err).to.not.be.null;
+                expect(err).to.have.status(422);
+                done();
+            });
+    });
+
+    it('Username Already Exist Return Error', (done) => {
```

Comment:
Replace 'Username Already Exist Return Error' with 'Returns Error when Username is already exists'